homeassistant/components/media_player/firetv.py

```diff
@@ -195,60 +155,134 @@ def source_list(self):
-        }.get(self._firetv.state, STATE_UNKNOWN)
-
-        if self._state not in [STATE_OFF, STATE_UNKNOWN]:
-            self._running_apps = self._firetv.running_apps
-            self._current_app = self._firetv.current_app
-        else:
-            self._running_apps = None
-            self._current_app = None
-
+        try:
+            # Check if device is disconnected.
+            if not self._firetv._adb:
+                self._state = STATE_UNKNOWN
+                self._running_apps = None
+                self._current_app = None
+
+                # Try to connect
+                self._firetv.connect()
+
+            # Check if device is off.
+            elif not self._firetv._screen_on:
+                self._state = STATE_OFF
+                self._running_apps = None
+                self._current_app = None
+
+            # Check if screen saver is on.
+            elif not self._firetv._awake:
+                self._state = STATE_IDLE
+                self._running_apps = None
+                self._current_app = None
+
+            else:
+                # Get the running apps.
+                if self._get_sources:
+                    self._running_apps = self._firetv.running_apps()
+
+                # Get the current app.
+                if self._get_source:
+                    current_app = self._firetv.current_app
+                    if isinstance(current_app, dict)\
+                            and 'package' in current_app:
+                        self._current_app = current_app['package']
+                    else:
+                        self._current_app = current_app
+
+                    # Show the current app as the only running app.
+                    if not self._get_sources:
+                        if self._current_app:
+                            self._running_apps = [self._current_app]
+                        else:
+                            self._running_apps = None
+
+                    # Check if the launcher is active.
+                    if self._current_app in [PACKAGE_LAUNCHER,
+                                             PACKAGE_SETTINGS]:
+                        self._state = STATE_STANDBY
+
+                    # Check for a wake lock (device is playing).
+                    elif self._firetv._wake_lock:
+                        self._state = STATE_PLAYING
+
+                    # Otherwise, device is paused.
+                    else:
+                        self._state = STATE_PAUSED
+
+                # Don't get the current app.
+                elif self._firetv._wake_lock:
+                    # Check for a wake lock (device is playing).
+                    self._state = STATE_PLAYING
+                else:
+                    # Assume the devices is on standby.
+                    self._state = STATE_STANDBY
+
+        except:  # pylint: disable=bare-except
+            _LOGGER.error('%s', 'Update encountered an exception; will ' +
```

Comment:
`_LOGGER.error('static string')` works just fine, no need for string interpolation.

Reply:
Although including the exception message would be useful for anyone troubleshooting it at some later point.

Reply:
I only did it this way to avoid pylint errors. When I split the string into two and concatenated them with a + I got a `logging-not-lazy` complaint.

Reply:
It is not necessary to concatenate them with a +, it will work fine without it.